src/main/java/io/vertx/core/impl/launcher/commands/ClasspathHandler.java

```diff
@@ -0,0 +1,166 @@
+      }
+    }).toArray(URL[]::new);
+    return new URLClassLoader(urls, this.getClass().getClassLoader());
+  }
+
+  /**
+   * Creates a new instance of {@link VertxIsolatedDeployer}.
+   *
+   * @return the new instance.
+   */
+  protected synchronized Object newInstance() {
+    try {
+      classloader = (classpath == null || classpath.isEmpty()) ?
+          ClasspathHandler.class.getClassLoader() : createClassloader();
+      Class<?> clazz = classloader.loadClass("io.vertx.core.impl.launcher.commands.VertxIsolatedDeployer");
+      return clazz.newInstance();
+    } catch (Exception e) {
+      log.error("Failed to load or instantiate the isolated deployer", e);
+      throw new IllegalStateException(e);
+    }
+  }
+
+  /**
+   * Creates a new non-clustered vert.x instance.
+   *
+   * @param options the options
+   * @return the created instance
+   */
+  protected synchronized Vertx create(VertxOptions options) {
+    final ClassLoader originalClassLoader = Thread.currentThread().getContextClassLoader();
+    try {
+      Thread.currentThread().setContextClassLoader(classloader);
+      return Vertx.vertx(options);
+    } catch (Exception e) {
+      log.error("Failed to create the vert.x instance", e);
+    } finally {
+      Thread.currentThread().setContextClassLoader(originalClassLoader);
+    }
+    return null;
+  }
+
+  /**
+   * Creates a new clustered vert.x instance.
+   *
+   * @param options       the options
+   * @param resultHandler the result handler
+   */
+  protected synchronized void create(VertxOptions options, Handler<AsyncResult<Vertx>> resultHandler) {
+    final ClassLoader originalClassLoader = Thread.currentThread().getContextClassLoader();
+    try {
+      Thread.currentThread().setContextClassLoader(classloader);
+      Vertx.clusteredVertx(options, resultHandler);
+    } catch (Exception e) {
+      log.error("Failed to create the vert.x instance", e);
+    } finally {
+      Thread.currentThread().setContextClassLoader(originalClassLoader);
+    }
+  }
+
+  /**
+   * Deploys the given verticle using the given deployment options.
+   *
+   * @param verticle          the verticle
+   * @param vertx             the vert.x instance
+   * @param options           the deployment options
+   * @param completionHandler the completion handler
+   */
+  public synchronized void deploy(String verticle, Vertx vertx, DeploymentOptions options,
+                                  Handler<AsyncResult<String>> completionHandler) {
+    if (manager == null) {
+      manager = newInstance();
+    }
+
+    final ClassLoader originalClassLoader = Thread.currentThread().getContextClassLoader();
+    try {
+      Thread.currentThread().setContextClassLoader(classloader);
+      Method method = manager.getClass().getMethod("deploy", String.class, Vertx.class, DeploymentOptions.class,
+          Handler.class);
+      method.invoke(manager, verticle, vertx, options, completionHandler);
+    } catch (Exception e) {
```

Comment:
perhaps handle InvocationException specially to unwrap and log the cause directly